src/client/Microsoft.Identity.Client/Platforms/net5win/Net5WamInterop.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.WindowsRuntime;
+using Windows.Foundation;
+using Windows.Security.Authentication.Web.Core;
+using Windows.Security.Credentials;
+using Windows.UI.ApplicationSettings;
+using WinRT;
+
+namespace Microsoft.Identity.Client.Platforms.net5win
+{
+    internal static class WebAuthenticationCoreManagerInterop
+    {
+        public static IAsyncOperation<WebTokenRequestResult> RequestTokenForWindowAsync(IntPtr hWnd, WebTokenRequest request)
+        {
+            IWebAuthenticationCoreManagerInterop webAuthenticationCoreManagerInterop =
+                WebAuthenticationCoreManager.As<IWebAuthenticationCoreManagerInterop>();
+            Guid guid = WinRT.GuidGenerator.CreateIID(typeof(IAsyncOperation<WebTokenRequestResult>));
+
+            var requestPtr = MarshalInspectable<WebTokenRequest>.FromManaged(request);
+
+            webAuthenticationCoreManagerInterop.RequestTokenForWindowAsync(
+               hWnd,
+               requestPtr,
+               ref guid,
+               out IntPtr result);
+
+            return MarshalInterface<IAsyncOperation<WebTokenRequestResult>>.FromAbi(result);
+        }
+
+        public static IAsyncOperation<WebTokenRequestResult> RequestTokenWithWebAccountForWindowAsync(
```

Comment:
>WebAccount [](start = 77, length = 10)

what is a WebAccount?

Reply:
WAM has it's own object model, and WebAccount is the equivalent of an MSAL.Account. The actual object is defined in WAM  API and exposed via WinRT. You will not see WebAccount mentioned outside WAM files.

There is extensive logic to match an IAccount to a WebAccount and vice-versa.

Reply:
oh right...you already explained this to me in the WAM PR...i won't ask a third time ;)

---
In reply to: [537352096](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/2279#discussion_r537352096) [](ancestors = 537352096)